snappi/snappicommon.py

```diff
@@ -141,15 +237,34 @@ def parent(self):
         return self._parent
         
     def _get_property(self, name, default_value=None, parent=None, choice=None):
-        if name not in self._properties or self._properties[name] is None:
-            if isinstance(default_value, type) is True:
-                self._properties[name] = default_value(parent=parent, choice=choice)
+        if name in self._properties and self._properties[name] is not None:
+            return self._properties[name]
+        if isinstance(default_value, type) is True:
+            if 'choice' in dir(self):
+                self._properties['choice'] = name
+            self._properties[name] = default_value(parent=parent, choice=choice)
+
+            if '_DEFAULTS' in dir(self._properties[name]) and\
+                'choice' in self._properties[name]._DEFAULTS:
+                # this node is used to set the if the choice has default.
+                # but seeing a side effect, if user sets other than default,
+                # in serialization we would see the default choice as well.
+                # currently there is not impact but it might deceive the perception
+                getattr(self._properties[name], self._properties[name]._DEFAULTS['choice'])
+        else:
+            if default_value is None and name in self._DEFAULTS:
+                if 'choice' in dir(self):
```

Comment:
had raised this previously - we should only set `self._properties['choice'] = name` if `name` is a valid enum.

Reply:
We had already discussed in meeting other day. We cannot assign any non-primitive data types, all are readonly properties.
f.size can't be set as there is no setter def defined
same as above

Reply:
coded changes done in latest revision